util/chunk/row_container_test.go

```diff
@@ -204,3 +205,48 @@ func (r *rowContainerTestSuite) TestSortedRowContainerSortSpillAction(c *check.C
 	err = rc.Reset()
 	c.Assert(err, check.IsNil)
 }
+
+func (r *rowContainerTestSerialSuite) TestActionBlocked(c *check.C) {
+	sz := 4
+	fields := []*types.FieldType{types.NewFieldType(mysql.TypeLonglong)}
+	rc := NewRowContainer(fields, sz)
+
+	chk := NewChunkWithCapacity(fields, sz)
+	for i := 0; i < sz; i++ {
+		chk.AppendInt64(0, int64(i))
+	}
+	var tracker *memory.Tracker
+	var err error
+	// Case 1
+	tracker = rc.GetMemTracker()
+	tracker.SetBytesLimit(1450)
+	ac := rc.ActionSpill()
+	tracker.FallbackOldAndSetNewAction(ac)
+	for i := 0; i < 10; i++ {
+		err = rc.Add(chk)
+		c.Assert(err, check.IsNil)
+	}
+
+	ac.cond.L.Lock()
+	for ac.cond.status == todoStatus ||
+		ac.cond.status == doingStatus {
+		ac.cond.Wait()
+	}
+	ac.cond.L.Unlock()
```

Comment:
Not only the `doneStatus` can be checked here, but also the memory usage and disk usage.